bundles/CoreBundle/DependencyInjection/Configuration.php

```diff
@@ -769,6 +769,28 @@ private function addObjectsNode(ArrayNodeDefinition $rootNode)
                                 ->end()
                             ->end()
                         ->end()
+                        ->arrayNode('custom_layout')
+                            ->addDefaultsIfNotSet()
+                            ->children()
+                                ->arrayNode('definitions')
+                                ->normalizeKeys(false)
+                                    ->prototype('array')
+                                        ->children()
+                                            ->scalarNode('id')->end()
+                                            ->scalarNode('name')->end()
+                                            ->scalarNode('description')->end()
+                                            ->integerNode('creationDate')->end()
+                                            ->integerNode('modificationDate')->end()
+                                            ->integerNode('userOwner')->end()
+                                            ->integerNode('userModification')->end()
+                                            ->scalarNode('classId')->end()
+                                            ->integerNode('default')->end()
+                                            ->variableNode('layoutDefinitions')->end()
```

Comment:
Sooner or later also Pimcore Class Definitions make sense to be moved to a Symfony Configuration. (not the container config, but a config though). Since Pimcore internally needs a lot of different fields within the layoutDefinitions, it makes sense to better describe what is allowed here

Reply:
we can't define allowed types in layoutDefinitions as this stores serialized layout data for BC reasons. please see https://github.com/pimcore/pimcore/pull/12779/files#diff-7657572963ff89400760ba6fabcfe4ae198543d224132bc7ced6e562504bd799R208. 

yes, it absolutely make sense when we decide to move class definitions to Symfony config.

Reply:
is it necessary to serialize it?

Reply:
yes, as the storage can also be settingsStore db table, where the data is stored in a single column.

Reply:
shouldn't the settings-store then be responsible for Serialization and De-Serialization?